.github/workflows/build-and-publish-rule-book.yaml

```diff
@@ -126,17 +138,84 @@ jobs:
           post_compile: echo "icu_locale \"${{ matrix.language }}\"" >> index_style.ist && upmendex -s index_style main_${{ matrix.language }} && sed -i 's@\(\\noindent\\textbf{\)\(.\)@\\noindent\\textbf{\2}@g' main_${{ matrix.language }}.ind && ${{ matrix.luaengine && 'lualatex' || 'pdflatex' }} --shell-escape main_${{ matrix.language }}.tex
           root_file: main_${{ matrix.language }}.tex
 
+      - name: Check number of printable pages
+        id: check_printable_pages
+        if: ${{ matrix.language == 'en' }}
+        run: |
+          NUM_OF_PAGES=$(pdfinfo main_${{ matrix.language }}.pdf | grep Pages | grep -Po "\d+")
+          echo "NUM_OF_PRINTABLE_PAGES=${NUM_OF_PAGES}" >> "$GITHUB_OUTPUT"
+
+      - name: Check if printable page number matches
+        if: ${{ matrix.language == 'en' }}
+        run: |
+          PAGES_DIGITAL="${{ steps.check_pages.outputs.num_of_pages }}"
+          PAGES_PRINTABLE="${{ steps.check_printable_pages.outputs.num_of_printable_pages }}"
+          echo "Number of pages in digital:   ${PAGES_DIGITAL}"
+          echo "Number of pages in printable: ${PAGES_PRINTABLE}"
+          if [[ $((${PAGES_DIGITAL}+1)) != ${PAGES_PRINTABLE} ]]; then
+            echo "Pages in digital and printable versions differ by too much. See above. Please investigate this."
+            exit 1
+          fi
+
       - name: Move file
-        if: github.event_name != 'pull_request'
         run: |
+          mkdir ${{ matrix.language }}
           mv main_${{ matrix.language }}.pdf ${{ matrix.language }}/printable_${{ matrix.language }}.pdf
 
-      - name: Publish ${{ matrix.language }} in artifacts repository
-        if: github.event_name != 'pull_request'
-        uses: peaceiris/actions-gh-pages@v4
+      - uses: actions/upload-artifact@v4
+        if: ${{ matrix.language != 'en' && github.event_name == 'pull_request' }}
+        with:
+          name: printable_${{ matrix.language }}
+          path: ${{ matrix.language }}/printable_${{ matrix.language }}.pdf
+          retention-days: ${{ env.ARTIFACT_RETENTION_DAYS }}
+      #
+      # - name: Publish ${{ matrix.language }} in artifacts repository
+      #   if: github.event_name != 'pull_request'
+      #   uses: peaceiris/actions-gh-pages@v4
+      #   with:
+      #     deploy_key: ${{ secrets.SSH_DEPLOY_KEY_BUILD_ARTIFACTS }}
+      #     external_repository: qwrtln/Homm3BG-build-artifacts
+      #     publish_branch: ${{ matrix.language }}
+      #     publish_dir: ./${{ matrix.language }}
+      #     force_orphan: true
+
+  check_pages:
+    runs-on: ubuntu-latest
+    needs: build
+    strategy:
+      fail-fast: false
+      matrix:
+        language: ["pl", "es", "fr", "ru"]
+
+    steps:
+      - name: Install pdfinfo
+        run: sudo apt-get install -y poppler-utils
+
+
+      - name: Download PDFs
```

Comment:
I think this need to explicitly depend on the job above, so that github doesn't run it too early

Reply:
It already depends.